pkg/operator/prunecontroller/prune_controller.go

```diff
@@ -64,24 +53,15 @@ func NewPruneController(
 		podGetter:      podGetter,
 		podInformer:    informers.InformersFor(targetNamespace).Core().V1().Pods(),
 		secretInformer: informers.InformersFor(targetNamespace).Core().V1().Secrets(),
-		eventRecorder:  eventRecorder.WithComponentSuffix("prune-controller"),
-
-		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "PruneController"),
 	}
 
-	c.podInformer.Informer().AddEventHandler(c.eventHandler())
-	c.secretInformer.Informer().AddEventHandler(c.eventHandler())
-
-	c.cachesToSync = append(
-		c.cachesToSync,
-		c.podInformer.Informer().HasSynced,
-		c.secretInformer.Informer().HasSynced,
-	)
-
-	return c
+	return factory.New().WithInformers(
+		c.podInformer.Informer(),
+		c.secretInformer.Informer(),
+	).WithSync(c.sync).ToController("PruneController", eventRecorder.WithComponentSuffix("prune-controller"))
```

Comment:
@mfojtik do I need a custom `resync` period ?

Reply:
I think that `sync` will also be trigger by the internal informers `resync` period